AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthConfirmSignUpTask.swift

```diff
@@ -7,110 +7,43 @@
-                    continuation.resume()
-                default:
-                    let error = AuthError.invalidState("Auth state must be signed out to signup", "", nil)
-                    continuation.resume(throwing: error)
-                }
-            }
-        }
 
-    }
-    
-    func doConfirmSignUp() async throws -> AuthSignUpResult {
-        try await withCheckedThrowingContinuation { [weak self] (continuation: CheckedContinuation<AuthSignUpResult, Error>) in
-            stateMachineToken = authStateMachine.listen { state in
-                guard case .configured(let authNState, _) = state else {
-                    return
-                }
-
-                switch authNState {
-                case .signingUp(let signUpState):
-                    switch signUpState {
-                    case .signedUp:
-                        continuation.resume(returning: AuthSignUpResult(.done))
-                    case .error(let error):
-                        continuation.resume(throwing: error.authError)
-                    default:
-                        break
-                    }
-                default:
-                    break
-                }
-            } onSubscribe: { [weak self] in
-                guard let self = self else { return }
-                self.sendConfirmSignUpEvent(continuation)
-            }
-        }
-    }
-    
-    private func sendConfirmSignUpEvent(_ continuation: CheckedContinuation<AuthSignUpResult, Error>) {
-        guard !request.code.isEmpty else {
-               let error = AuthError.validation(
-                AuthPluginErrorConstants.confirmSignUpCodeError.field,
-                AuthPluginErrorConstants.confirmSignUpCodeError.errorDescription,
-                AuthPluginErrorConstants.confirmSignUpCodeError.recoverySuggestion, nil)
-            continuation.resume(throwing: error)
-            return
-        }
-        let confirmSignUpData = ConfirmSignUpEventData(
-            username: request.username,
-            confirmationCode: request.code)
-        let event = SignUpEvent(eventType: .confirmSignUp(confirmSignUpData))
-        authStateMachine.send(event)
-    }
-    
-    private func cancelToken() {
-        if let token = stateMachineToken {
-            authStateMachine.cancel(listenerToken: token)
+            let client = try userPoolEnvironment.cognitoUserPoolFactory()
+            let input = ConfirmSignUpInput(username: request.username,
+                                           confirmationCode: request.code,
+                                           environment: userPoolEnvironment)
+            _ = try await client.confirmSignUp(input: input)
+            return AuthSignUpResult(.done)
+        } catch let error as AuthError {
+            throw error
+        } catch let error as AuthErrorConvertible {
+            throw error.authError
+        } catch let error {
+            let error = AuthError.configuration(
```

Comment:
is this the correct AuthError and description?

Reply:
Used the similar error from other operations like `AWSAuthFetchUserAttributesOperation`